inference-engine/src/auto_plugin/auto_plugin.cpp

```diff
@@ -24,20 +25,142 @@ namespace {
+        }
+
+        std::vector<DeviceInformation> CPU;
+        std::vector<DeviceInformation> GPU;
+
         for (auto& item : metaDevices) {
             if (item.deviceName.find("CPU") == 0) {
-              return item;
+                CPU.push_back(item);
+                continue;
+            }
+            if (item.deviceName.find("GPU") == 0) {
+                GPU.push_back(item);
+                continue;
+            }
+        }
+
+        auto networkPrecision = GetNetworkPrecision(network);
+        auto getCap = [&](std::string&& substr){
+        auto capability = std::find_if(optCap.begin(), optCap.end(),
+                                    [&](const std::string& c)->bool{ return (c.find(substr) != std::string::npos);});
+            return capability;
+        };
+
+        if (CPU.empty() && GPU.empty()) {
+            IE_THROW(NotFound) << "No available device found";
+        }
+
+        std::sort(GPU.begin(), GPU.end(), [](const DeviceInformation& a, const DeviceInformation& b)->bool{return b.deviceName < a.deviceName;});
+
+        if (!GPU.empty()) {
+            auto capability = getCap("GPU");
+            if (capability != optCap.end() && capability->find(networkPrecision) != std::string::npos) {
+                return GPU[0];
             }
         }
-        IE_THROW(NotFound) << "No available device could be used";
+        if (CPU.empty()) {
+            IE_THROW() << "Cannot select any device";
+        }
+        return CPU[0];
     }
 }  // namespace
 
 AutoInferencePlugin::AutoInferencePlugin() {
     _pluginName = "AUTO";
 }
 
+IE::ExecutableNetwork AutoInferencePlugin::LoadNetwork(const std::string &modelPath,
+                                                       const std::map<std::string, std::string> &config) {
+    if (GetCore() == nullptr) {
+        IE_THROW() << "Please, work with AUTO device via InferencEngine::Core object";
+    }
+
+    auto fullConfig = mergeConfigs(_config, config);
+    auto metaDevices = GetDeviceChoice(fullConfig);
+
+    DeviceInformation selectedDevice;
+    IE::ExecutableNetwork executableNetwork;
+    while (!metaDevices.empty()) {
+        selectedDevice = SelectDevice(metaDevices);
+        try {
+            executableNetwork = GetCore()->LoadNetwork(modelPath, selectedDevice.deviceName, config);
+            break;
+        } catch (...) {
+            auto eraseDevice = std::find_if(metaDevices.begin(), metaDevices.end(),
+                                            [=](const DeviceInformation &d) -> bool { return d.deviceName == selectedDevice.deviceName; });
+            if (eraseDevice == metaDevices.end()) {
+                IE_THROW() << "Didn't find the selected device name";
+            }
+            metaDevices.erase(eraseDevice);
+            executableNetwork = {};
+        }
+    }
+    if (!executableNetwork) {
+        IE_THROW() << "Failed to load network by AUTO plugin";
+    }
+
+    return executableNetwork;
```

Comment:
we need to wrap it with `AutoExecutableNetwork`..

Reply:
It seems we could not able to wrap it with `AutoExecutableNetwork`, because the return type is `IE::ExecutableNetwork `, but `AutoExecutableNetwork` is derived from `ExecutableNetworkInternal`.

Reply:
ok, we are going to fix it here https://github.com/openvinotoolkit/openvino/pull/5645